k2/python/k2/fsa_algo.py

```diff
@@ -17,25 +18,58 @@
 # Note: look also in autograd.py, differentiable operations may be there.
 
 
-def linear_fsa(symbols: Union[List[int], List[List[int]]]) -> Fsa:
-    '''Construct an linear FSA from symbols.
+def linear_fsa(labels: Union[List[int], List[List[int]]]) -> Fsa:
+    '''Construct an linear FSA from labels.
 
     Note:
       The scores of arcs in the returned FSA are all 0.
 
     Args:
-      symbols:
+      labels:
         A list of integers or a list of list of integers.
 
     Returns:
-      An FSA if the input is a list of integers.
+      An FSA if the labels is a list of integers.
       A vector of FSAs if the input is a list of list of integers.
     '''
-    ragged_arc = _k2.linear_fsa(symbols)
+    ragged_arc = _k2.linear_fsa(labels)
     fsa = Fsa(ragged_arc)
     return fsa
 
 
+def linear_fst(labels: Union[List[int], List[List[int]]],
+               aux_labels: Union[List[int], List[List[int]]]) -> Fsa:
+    '''Construct an linear FST from labels and its corresponding
+    auxiliary labels.
+
+    Note:
+      The scores of arcs in the returned FST are all 0.
+
+    Args:
+      labels:
+        A list of integers or a list of list of integers.
+      aux_labels:
+        A list of integers or a list of list of integers.
+
+    Returns:
+      An FST if the labels is a list of integers.
+      A vector of FSTs if the input is a list of list of integers.
+    '''
+    ragged_arc = _k2.linear_fsa(labels)
+    if isinstance(labels[0], List):
+        assert isinstance(aux_labels[0],
+                          List), 'aux_labels and labels do not match.'
+        flattened_labels = []
+        for aux in aux_labels:
+            flattened_labels.extend(aux + [-1])  # -1 == kFinalSymbol
+        aux_labels = torch.IntTensor(flattened_labels)
+    else:
+        aux_labels.append(-1)  # -1 == kFinalSymbol
```

Comment:
This changes the input `aux_labels` **in-place**. 

I would suggest making a copy of it.